uber/site_sections/budget.py

```diff
@@ -166,7 +166,15 @@ def generate_promo_codes(
                 params['code'] = code
                 promo_codes.append(PromoCode().apply(params))
 
+            message = check_all(promo_codes)
+            if message:
+                result['message'] = message
```

Comment:
Be sure to check to make sure that doesn't cause unwanted behavior with clearing the form on validation error -- your way may be much better if that's the case.